### PR TITLE
[android] Use PwoMetadata in DiscoveryService

### PR DESCRIPTION
**Pull request by cco3**
Simplify the code some by using the PwoMetadata class in the
UriBeaconDiscoveryService.

### CODE REVIEW

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UriBeaconDiscoveryService.java [hunk 1]
```diff
@@ -356,14 +364,20 @@ private void updateNotifications() {
   }
 
   private ArrayList<String> getSortedBeaconsWithMetadata() {
-    ArrayList<String> unSorted = new ArrayList<>(mDeviceAddressToUrl.size());
+    ArrayList<PwoMetadata> pwoMetadataList = new ArrayList<>(mUrlToPwoMetadata.size());
```
Comment (matthewsibigtroth) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UriBeaconDiscoveryService.java]: do we need the size() call here?

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UriBeaconDiscoveryService.java [hunk 2]
```diff
@@ -356,14 +364,20 @@ private void updateNotifications() {
...
+      PwoMetadata pwoMetadata = mUrlToPwoMetadata.get(mDeviceAddressToUrl.get(key));
+      if (pwoMetadata != null && pwoMetadata.hasUrlMetadata() && pwoMetadata.hasBleMetadata()) {
+        pwoMetadataList.add(pwoMetadata);
       }
     }
-    Collections.sort(unSorted, new MetadataComparator(mUrlToUrlMetadata));
-    return unSorted;
+    Collections.sort(pwoMetadataList);
+
+    ArrayList<String> deviceAddresses = new ArrayList<>(mUrlToPwoMetadata.size());
```
Comment (matthewsibigtroth) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UriBeaconDiscoveryService.java]: do we need the size() call here?

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UriBeaconDiscoveryService.java (hunk 1 above)
Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UriBeaconDiscoveryService.java]: No, it's just an algorithmic optimization.  same for the below.

**Comment by matthewsibigtroth** [comment]
Two minor comments. Otherwise LGTM.

**Comment by cco3** [comment]
If that's it, I'm going to leave the size parameter and submit.

**Pull request merged**